common/component/kafka/metadata.go

```diff
@@ -244,6 +246,9 @@ func (k *Kafka) getKafkaMetadata(meta map[string]string) (*KafkaMetadata, error)
 		if m.AWSRegion == "" {
 			return nil, errors.New("missing AWS region property 'awsRegion' for authType 'awsiam'")
 		}
+		if m.AWSIamRoleArn == "" && m.AWSSecretKey == "" && m.AWSAccessKey == "" {
```

Comment:
Access key and secret key can actually be empty and its a supported feature, so the aws sdk picks the creds automatically from  other sources such as the environment. This new if condition breaks that feature

Reply:
cc @eunicecompra

Reply:
Yep! gotcha. I will remove this check.